envoy/upstream/load_balancer.h

```diff
@@ -28,6 +28,35 @@ namespace Upstream {
 
 using ClusterProto = envoy::config::cluster::v3::Cluster;
 
+/*
+ * A handle to allow cancelation of asynchronous host selection.
+ * If chooseHost returns a HostSelectionResponse with an AsyncHostSelection
+ * handle, and the endpoint does not wish to receive onAsyncHostSelction call,
+ * it must call cancel() on the provided handle.
+ */
+struct AsyncHostSelection {
```

Comment:
It looks like we use this below in a shared_ptr. In that case, should we disable copying this (and perhaps make it a class, since it has no members and does have logic, which the style reserves for classes)?

Also, maybe this should have the suffix `Handle`. I think that would better match the purpose of this class.

Reply:
class and handle done